src/node_buffer.cc

```diff
@@ -18,6 +18,12 @@
     if (!(r)) return env->ThrowRangeError("out of range index");            \
   } while (0)
 
+#define CHECK_IS_BUFFER(env, obj)                                           \
+  do {                                                                      \
```

Comment:
I am trying to understand this. When will the loop break? It appears that this infinitely checks if `obj`  HasInstance